flaky/_flaky_plugin.py

```diff
@@ -212,48 +228,37 @@ def _handle_test_success(self, test):
         :rtype:
             `bool`
         """
-        _, _, name = self._get_test_declaration_callable_and_name(test)
-        current_runs = self._get_flaky_attribute(
-            test,
-            FlakyNames.CURRENT_RUNS
-        )
-        if current_runs is None:
+        try:
+            _, _, name = self._get_test_declaration_callable_and_name(test)
+        except AttributeError:
             return False
-        current_runs += 1
-        current_passes = self._get_flaky_attribute(
-            test,
-            FlakyNames.CURRENT_PASSES
-        )
-        current_passes += 1
-        self._set_flaky_attribute(
-            test,
-            FlakyNames.CURRENT_RUNS,
-            current_runs
-        )
-        self._set_flaky_attribute(
-            test,
-            FlakyNames.CURRENT_PASSES,
-            current_passes
-        )
-        flaky = self._get_flaky_attributes(test)
-        need_reruns = not self._has_flaky_test_succeeded(flaky)
-        if self._flaky_success_report:
+        need_reruns = self._should_handle_test_success(test)
+
+        if self._has_flaky_attributes(test):
+            flaky = self._get_flaky_attributes(test)
             min_passes = flaky[FlakyNames.MIN_PASSES]
-            self._stream.writelines([
-                ensure_unicode_string(name),
-                ' passed {0} out of the required {1} times. '.format(
-                    current_passes,
-                    min_passes,
-                ),
-            ])
-            if need_reruns:
-                self._stream.write(
-                    'Running test again until it passes {0} times.\n'.format(
+            passes = flaky[FlakyNames.CURRENT_PASSES] + 1
+            runs = flaky[FlakyNames.CURRENT_RUNS] + 1
```

Comment:
Should we have helper methods for:
- Changing flaky attributes to increase current_runs.
- Changing flaky attributes to increase current_runs and current_passes.

Then we don't need to duplicate code.